robot/client/client.go

```diff
@@ -71,17 +72,87 @@ type RobotClient struct {
 	closeContext context.Context
 }
 
+var exemptFromConnectionCheck = map[string]bool{
+	"/proto.rpc.webrtc.v1.SignalingService/Call":                 true,
+	"/proto.rpc.webrtc.v1.SignalingService/CallUpdate":           true,
+	"/proto.rpc.webrtc.v1.SignalingService/OptionalWebRTCConfig": true,
+	"/proto.rpc.v1.AuthService/Authenticate":                     true,
+}
+
+func needsConnectionCheck(method string) bool {
+	return !exemptFromConnectionCheck[method]
+}
+
+const readWriteOnClosedPipeErrorMsg = "read/write on closed pipe"
+
+func isRWOnClosedPipeError(err error) bool {
+	if err == nil {
+		return false
+	}
+	return strings.Contains(err.Error(), readWriteOnClosedPipeErrorMsg)
```

Comment:
I think this was originally my fault, but think we can use `io. ErrClosedPipe` instead